package/MDAnalysis/coordinates/base.py

```diff
@@ -1293,13 +1294,20 @@ def check_slice_indices(self, start, stop, step):
             range(10)[10:-1:-1]  # []
 
         """
-        for var, varname in (
-                (start, 'start'),
-                (stop, 'stop'),
-                (step, 'step')
-        ):
-            if not (isinstance(var, int) or (var is None)):
+
+        slice_dict = {'start': start, 'stop': stop, 'step': step}
```

Comment:
I like the fact that this check is now at the Reader level so that the analysis code does not have to worry about ints.